Python/Batch/sample4_job_scheduler.py

```diff
@@ -0,0 +1,233 @@
+    """Executes the sample with the specified configurations.
+
+    :param global_config: The global configuration to use.
+    :type global_config: `configparser.ConfigParser`
+    :param sample_config: The sample specific configuration to use.
+    :type sample_config: `configparser.ConfigParser`
+    """
+    # Set up the configuration
+    batch_account_key = global_config.get('Batch', 'batchaccountkey')
+    batch_account_name = global_config.get('Batch', 'batchaccountname')
+    batch_service_url = global_config.get('Batch', 'batchserviceurl')
+
+    storage_account_key = global_config.get('Storage', 'storageaccountkey')
+    storage_account_name = global_config.get('Storage', 'storageaccountname')
+    storage_account_suffix = global_config.get(
+        'Storage',
+        'storageaccountsuffix')
+
+    should_delete_container = sample_config.getboolean(
+        'DEFAULT',
+        'shoulddeletecontainer')
+    should_delete_job_schedule = sample_config.getboolean(
+        'DEFAULT',
+        'shoulddeletejobschedule')
+    pool_vm_size = sample_config.get(
+        'DEFAULT',
+        'poolvmsize')
+    pool_vm_count = sample_config.getint(
+        'DEFAULT',
+        'poolvmcount')
+
+    # Print the settings we are running with
+    common.helpers.print_configuration(global_config)
+    common.helpers.print_configuration(sample_config)
+
+    credentials = batchauth.SharedKeyCredentials(
+        batch_account_name,
+        batch_account_key)
+
+    batch_client = batch.BatchServiceClient(
+        credentials,
+        batch_url=batch_service_url)
+
+    block_blob_client = azureblob.BlockBlobService(
+        account_name=storage_account_name,
+        account_key=storage_account_key,
+        endpoint_suffix=storage_account_suffix)
+
+    batch_client.config.retry_policy.retries = 5
+    job_schedule_id = common.helpers.generate_unique_resource_name(
+        "JobScheduler")
+
+    try:
+        create_job_schedule(
+            batch_client,
+            job_schedule_id,
+            pool_vm_size,
+            pool_vm_count,
+            block_blob_client)
+
+        print("Start time: ", _START_TIME)
+        print("Delete time: ", _END_TIME)
+
+        recent_job = common.helpers.wait_for_job_under_job_schedule(
+            batch_client,
+            job_schedule_id,
+            datetime.timedelta(minutes=5))
+
+        common.helpers.wait_for_tasks_to_complete(
+            batch_client,
+            recent_job,
+            datetime.timedelta(minutes=25))
+
+        tasks = batch_client.task.list(recent_job)
+        task_ids = [task.id for task in tasks]
+
+        common.helpers.print_task_output(batch_client, recent_job,
+                                         task_ids)
+
+    except batchmodels.BatchErrorException as e:
+        for x in e.error.values:
+            print("BatchErrorException: ", x)
+
+    finally:
+        common.helpers.wait_for_job_schedule_to_complete(
```

Comment:
I dont think that this should be in the finally block - shouldn't it be up above in the happy-path?